src/LiveQuery/ParseLiveQueryServer.js

```diff
@@ -362,17 +373,28 @@ class ParseLiveQueryServer {
             // Then get the user's roles
           var rolesQuery = new Parse.Query(Parse.Role);
           rolesQuery.equalTo("users", user);
-          return rolesQuery.find({useMasterKey:true});
+
+            // fallback to direct query
+          if (!(this.cacheController instanceof RedisCacheAdapter)) {
```

Comment:
@Oziabr please help me understand - do the unit tests still run because of this fallback? Or do you pass the cacheController into the tests?